### PR TITLE
Add documentation for test helpers functions

### PR DESCRIPTION
**Pull request by mrKisaoLamb**
With this changes documentation of all test helpers will be visible here: https://pkg.go.dev/github.com/openstack-k8s-operators/lib-common/modules/test@v0.0.0-20230510145530-3c8e9179fb6e/helpers

### CODE REVIEW

File: modules/test-operators/helpers/database.go [hunk 1]
```diff
@@ -49,7 +49,23 @@ func (tc *TestHelper) CreateDBService(namespace string, mariadbCRName string, sp
 	return types.NamespacedName{Name: serviceName, Namespace: namespace}
 }
 
-// DeleteDBService -
+// DeleteDBService - deletes a database service from the Kubernetes cluster.
```
Comment (gibizer) [inline modules/test-operators/helpers/database.go:52]: you can drop the ` - `. I think it was just a placeholder

File: modules/test-operators/helpers/database.go [hunk 2]
```diff
@@ -67,7 +83,17 @@ func (tc *TestHelper) DeleteDBService(name types.NamespacedName) {
...
 
-// GetMariaDBDatabase -
+// GetMariaDBDatabase - retrieves a MariaDBDatabase resource from the Kubernetes cluster.
+//
+// The function creates a new MariaDBDatabase pointer and uses the Get function of the Kubernetes client to fetch the MariaDBDatabase.
+// If the MariaDBDatabase cannot be immediately retrieved, the function will retry for a duration specified by tc.Timeout,
+// checking every tc.Interval, until the MariaDBDatabase is successfully retrieved or the timeout duration is reached.
+// The function asserts that the retrieval operation should eventually succeed.
+//
+// The function returns a pointer to the retrieved MariaDBDatabase.
```
Comment (gibizer) [inline modules/test-operators/helpers/database.go:93]: This also feel chatty to me and very implementation specific. Also repeating things that already clear from the signature, like the return type.

For me the following would be enough:

GetMariaDBDatabase waits for and retrieves a MariaDBDatabase resource from the Kubernetes cluster

File: modules/test-operators/helpers/database.go [hunk 3]
```diff
@@ -76,14 +102,28 @@ func (tc *TestHelper) GetMariaDBDatabase(name types.NamespacedName) *mariadbv1.M
...
 
-// SimulateMariaDBDatabaseCompleted -
+// SimulateMariaDBDatabaseCompleted -  simulates a completed state for a MariaDBDatabase resource in a Kubernetes cluster.
+//
+// The function retrieves the MariaDBDatabase using the GetMariaDBDatabase method and then modifies the Completed property of the MariaDBDatabase's status.
+// It sets the Completed property to true, indicating that the database operation is completed.
+// The function then updates the status of the MariaDBDatabase in the cluster using the Status().Update method of the Kubernetes client.
+// If the update cannot be immediately completed, the function will retry for a duration specified by tc.Timeout,
+// checking every tc.Interval, until the update is successful or the timeout duration is reached.
+// The function asserts that the update operation should eventually succeed.
```
Comment (gibizer) [inline modules/test-operators/helpers/database.go:119]: ditto, this is chatty. We don't need to repeat the golang code in english.

File: modules/test/helpers/configmap.go [hunk 4]
```diff
@@ -44,7 +56,7 @@ func (tc *TestHelper) ListConfigMaps(namespace string) *corev1.ConfigMapList {
 	return cms
 }
 
-// CreateEmptyConfigMap -
+// CreateEmptyConfigMap - Deprecated and unused
```
Comment (gibizer) [inline modules/test/helpers/configmap.go:59]: Yepp, I think I transformed the last usage of this out of nova-operator recently. Lets delete it

File: modules/test/helpers/deployment.go [hunk 5]
```diff
@@ -31,15 +46,15 @@ func (tc *TestHelper) GetDeployment(name types.NamespacedName) *appsv1.Deploymen
...
+// ListDeployments - not used
 func (tc *TestHelper) ListDeployments(namespace string) *appsv1.DeploymentList {
 	deployments := &appsv1.DeploymentList{}
 	gomega.Expect(tc.K8sClient.List(tc.Ctx, deployments, client.InNamespace(namespace))).Should(gomega.Succeed())
 
 	return deployments
 }
 
-// SimulateDeploymentReplicaReady -
+// SimulateDeploymentReplicaReady - not used
```
Comment (gibizer) [inline modules/test/helpers/deployment.go:57]: Here I'm not sure. I might keep it as we might need it in the future.

File: modules/test/helpers/service.go [hunk 6]
```diff
@@ -22,7 +22,7 @@ import (
 	"k8s.io/apimachinery/pkg/types"
 )
 
-// CreateService -
+// CreateService - never used(?)
```
Comment (gibizer) [inline modules/test/helpers/service.go:25]: I think it might be used by https://github.com/openstack-k8s-operators/infra-operator/pull/39

**Comment by mrKisaoLamb** [comment]
The documentation has now been shortened, and any unused functions have been deleted

**Review by gibizer — COMMENTED** [review]
Looks good. But you need to rebase.

**Comment by mrKisaoLamb** [comment]
> Looks good. But you need to rebase.

Thanks! Done!

File: modules/test/helpers/statefulset.go
Comment (gibizer) [inline modules/test/helpers/statefulset.go:1]: The rebase pulled in new functions without doc `SimulateStatefulSetReplicaReadyWithPods`

Comment (mrKisaoLamb) [reply modules/test/helpers/statefulset.go:1]: done 4ba4b480db41b4d76afa512ffbf39239fac4492a

File: modules/test/helpers/deployment.go
[the remark this answers is not in the document]
Comment (mrKisaoLamb) [reply modules/test/helpers/deployment.go:1]: done 4ba4b480db41b4d76afa512ffbf39239fac4492a

**Review by gibizer — APPROVED** [review]

**Pull request merged**